test/unit/dom/hasChildNodes.js

```diff
@@ -0,0 +1,104 @@
+import create from '../../lib/create';
+import canPatchNativeAccessors from '../../../src/util/can-patch-native-accessors';
+
+describe('dom: firstElementChild', function () {
+  runTests('div');
+  runTests('slot');
+  runTests('host');
+  runTests('root');
+
+  function runTests(type) {
+    describe(`${type}: `, () => {
+      let host, root, slot, div, elem;
+
+      beforeEach(function () {
+        host = document.createElement('div');
+        root = host.attachShadow({ mode: 'open' });
+        slot = create('slot');
+
+        root.appendChild(slot);
+
+        div = document.createElement('div');
+
+        switch(type) {
+          case 'div':
+            elem = div;
+            break;
+          case 'slot':
+            elem = slot;
+            break;
+          case 'root':
+            root.innerHTML = '';
+            elem = root;
+            break;
+          default:
+            elem = host;
+        }
+      });
+
+      it(`should return false if there are no child nodes`, () => {
```

Comment:
can just use normal single-quotes rather than backticks for strings that don't contain vars (not sure if this is an airbnb rule, if it is then feel free to ignore for now)